cidrset/cidr_set.go

```diff
@@ -70,8 +78,15 @@ func NewCIDRSet(clusterCIDR *net.IPNet, subNetMaskSize int) (*CidrSet, error) {
 
 	isV6 := clusterCIDR.IP.To4() == nil
 	var maxCIDRs int
-	if (isV6) && (subNetMaskSize-clusterMaskSize > clusterSubnetMaxDiff) {
-		return nil, ErrCIDRSetSubNetTooBig
+	if isV6 {
+		if subNetMaskSize-clusterMaskSize > clusterSubnetMaxDiff {
+			return nil, ErrCIDRSetSubNetTooBig
+		}
+		if subNetMaskSize > maxSubNetMaskSizeIPv6 {
+			return nil, ErrSubNetMaskSizeInvalid
+		}
```

Comment:
I would reverse the order of the checks. There's no point in checking if the difference for the subnets if we can already tell the submask provided is `>128`.
 
```suggestion
		if subNetMaskSize > maxSubNetMaskSizeIPv6 {
			return nil, ErrSubNetMaskSizeInvalid
		}
		if subNetMaskSize-clusterMaskSize > clusterSubnetMaxDiff {
			return nil, ErrCIDRSetSubNetTooBig
		}
```

Reply:
done